resources/Microsoft.NET.SDK.ToolInstaller/Microsoft.NET.SDK.ToolInstaller.psm1

```diff
@@ -0,0 +1,404 @@
+                PackageId = $columns[0]
+                Version   = $columns[1]
+                Commands  = $columns[2]
+            }
+
+            # Add the custom object to the array
+            $inputObject += $customObject
+        }
+
+        return $inputObject
+    }
+}
+
+function Install-DotNetToolPackage
+{
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory, ValueFromPipelineByPropertyName)]
+        [string] $Name,
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [string] $Version,
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [bool] $PreRelease,
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [string] $ToolPath
+    )
+    
+    begin
+    {
+        function Get-DotNetToolArguments
+        {
+            param([string]$Name, [string]$Version, [bool]$PreRelease, [string]$ToolPath)
+
+            $string = $Name
+            
+            if (-not ([string]::IsNullOrEmpty($Version)))
+            {
+                $string += " --version $Version"
+            }
+
+            if ($PreRelease)
+            {
+                $string += " --prerelease"
+            }
+
+            if ($ToolPath)
+            {
+                $string += " --tool-path $ToolPath"
+            }
+            else 
+            {
+                $string += " --global"
+            }
+
+            $string += " --no-cache"
+
+            return $string
+        }
+    }
+    
+    process
+    {
+        $installArgument = Get-DotNetToolArguments @PSBoundParameters
+        $arguments = "tool install $installArgument --ignore-failed-sources"
+        Write-Verbose -Message "Installing dotnet tool package with arguments: $arguments"
+
+        Invoke-DotNet -Command $arguments
+    }
+}
+
+function Uninstall-DotNetToolPackage
+{
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory, ValueFromPipelineByPropertyName)]
+        [string]$Name
+    )
+
+    $arguments = "tool uninstall $Name --global" 
+    Write-Verbose -Message "Uninstalling dotnet tool package with arguments: $arguments"
+        
+    Invoke-DotNet -Command
+}
+
+function Invoke-DotNet
+{
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $Command
+    )
+
+    try
+    {
+        Invoke-Expression "& `"$DotNetCliPath`" $Command"
+    }
+    catch
+    {
+        throw "Executing dotnet.exe with {$Command} failed."
+    }
+}
+
+# Keeps the path of the code.exe CLI path.
+$DotNetCliPath = Get-DotNetPath
+
+#endregion Functions
+
+#region Classes
+<#
+.SYNOPSIS
+    This class is used to install and uninstall .NET SDK tools globally or use the tool path directory.
+#>
+[DSCResource()]
+class NETSDKToolInstaller
+{
+    [DscProperty(Key)]
+    [string] $PackageId
+
+    [DscProperty()]
+    [string] $Version
+
+    [DscProperty()]
+    [string[]] $Commands
+
+    [DscProperty()]
+    [bool] $PreRelease = $false
+
+    [DscProperty()]
+    [string] $ToolPath
```

Comment:
Do we need another dsc boolean property for `Global`

Reply:
I don't think so Ryan, please check latest commit as we default to global.